src/contexts/Api/index.tsx

```diff
@@ -163,8 +146,13 @@ export const APIProvider = ({ children }: { children: React.ReactNode }) => {
   };
 
   // connect function sets provider and updates active network.
-  const connect = async (_network: NetworkName, _isLightClient?: boolean) => {
-    const nodeEndpoint: Network = NETWORKS[_network];
+  const connect = async (
+    networkName: NetworkName,
+    _isLightClient?: boolean
+  ) => {
+    setConnectionStatus(ConnectionStatus.Connecting);
```

Comment:
Moved from [over there](https://github.com/Cardinal-Cryptography/aleph-zero-dashboard/pull/37/files#diff-de701a1090eab6a6ce998b10578ba8d16bd3d81a57a7caf3ea8d1d14141ab0a8L196) - it belongs in here, being tigtly coupled to something called _connect_.